### PR TITLE
blitz-auth: Fix webpack from following next-auth

### PR DESCRIPTION
**Pull request by siddhsuresh**
Closes: ?

### What are the changes and their implications?

in betav29, webpack follows `next-auth` with webpack resolve when the package name is passed directly.

- [x] Upgrade CI to use pnpm `latest`

## Bug Checklist

- [x] Changeset added (run `pnpm changeset` in the root directory)

### CODE REVIEW

**Comment by tordans** [comment]
Just a quick +1 for this PR. We tried migrating [our blitz app](https://github.com/FixMyBerlin/trassenscout) to the app folder this week but got stuck on errors that sound like they will be solved by this PR. We are using blitz-auth, not  next-auth for our app. We will try again once this is merged…

**Comment by siddhsuresh** [comment]
Hey @tordans sorry for that, we are working to merge this soon.

**Review by flybayer — APPROVED** [review]

**Comment by datner** [comment]
OK! Now let's publish this 😃

**Comment by variousred** [comment]
need this. ♥

**Comment by variousred** [comment]
Bummer, i thought this would fix this
```
node ➜ /workspace (devcontainer-clean) $ yarn  dev 
yarn run v1.22.19
$ blitz dev
Loaded env from /workspace/.env.development
Loaded env from /workspace/.env.local
Loaded env from /workspace/.env
/workspace/node_modules/next/dist/client/index.js
✔ Next.js was successfully patched with a React Suspense fix
✔ Routes manifest was successfully generated
- ready started server on 0.0.0.0:3000, url: http://localhost:3000
- error Failed to load next.config.js, see more info here https://nextjs.org/docs/messages/next-config-error
Error [ERR_PACKAGE_PATH_NOT_EXPORTED]: Package subpath './core/init' is not defined by "exports" in /workspace/node_modules/next-auth/package.json
    at new NodeError (node:internal/errors:387:5)
    at throwExportsNotFound (node:internal/modules/esm/resolve:365:9)
    at packageExportsResolve (node:internal/modules/esm/resolve:649:3)
    at resolveExports (node:internal/modules/cjs/loader:554:36)
    at Function.Module._findPath (node:internal/modules/cjs/loader:594:31)
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:1012:27)
    at /workspace/node_modules/next/dist/server/require-hook.js:189:36
    at Function.Module._load (node:internal/modules/cjs/loader:871:27)
    at Module.require (node:internal/modules/cjs/loader:1098:19)
    at require (node:internal/modules/cjs/helpers:108:18) {
  type: 'Error',
  code: 'ERR_PACKAGE_PATH_NOT_EXPORTED'
}
error Command failed with exit code 1.
```

**Comment by datner** [comment]
@variousred this looks entirely unrelated to the `next-auth` situation 🤔 
Do you have `"moduleResolution": "node16"` or `"nodenext"` or `"bundler"` in your tsconfig?

**Comment by variousred** [comment]
```    "moduleResolution": "nodenext",
```

**Comment by datner** [comment]
@variousred 
Yeah this is an issue with `next-auth`. They probably do not expect someone to use `next-auth/core/init` directly, so it is not part of the `"exports"` field as seen [here](https://github.com/nextauthjs/next-auth/blob/a79774f6e890b492ae30201f24b3f7024d0d7c9d/packages/next-auth/package.json#L29)

blitz can just patch them, but it's not that simple:
1. if blitz patches `next-auth` it will be required to _bundle it_ with `@blitzjs/auth` because patches don't magically apply 
2. it will not actually solve the issue, this will have to be done on `next-auth`s side. Which they might not want as we don't know if they consider this api to be "internal"
3. nextjs can't handle `"moduleResolution": "nodenext"`, or any of the other I named above. So even if this problem is solved, I don't think you'll be able to actually use your nextjs application.

If you _did_ manage to make nextjs work with `"moduleResolution": "nodenext"` and `"type": "module"` please let me know because this is considered unsolvable from the outside as far as next see it 😄 

tl;dr
switch to `"moduleResolution": "node"`

**Comment by variousred** [comment]
whats the "right" way to use blitz-auth?

**Comment by variousred** [comment]
linking to comments at https://github.com/blitz-js/blitz/issues/4171

**Comment by datner** [comment]
@variousred do you perhaps still use "type": "module" in your package json?

**Comment by variousred** [comment]
No

**Comment by variousred** [comment]
> whats the "right" way to use blitz-auth with a provider like SIWE, rainbowkit?
> 
> just following https://blitzjs.com/docs/next-auth and https://docs.login.xyz/integrations/nextauth.js

has anyone done this? is this not working generally?

**Comment by siddhsuresh** [comment]
Hey @variousred , we are currently in the process of working with next auth to fix this issue by exposing the required functions upstream. 
But even this will be made in a newer adapter for `@auth/core` rather than the current `next-auth`.
For your use right now and what works at the moment would be a manual patch for next-auth using `pnpm patch next-auth ` and add these exports in the package.json of the `next-auth` package.

Thank you for patience, we are working to make this more stable.

**Comment by variousred** [comment]
thanks @siddhsuresh much appreciated
